packages/compass-aggregations/src/components/pipeline-results-workspace/index.tsx

```diff
@@ -1,24 +1,134 @@
 import React from 'react';
 import { connect } from 'react-redux';
-import type { ConnectedProps } from 'react-redux';
+import type { Document } from 'mongodb';
+import {
+  IconButton,
+  css,
+  spacing,
+  Icon,
+  Button,
+  Body,
+  uiColors,
+  cx,
+} from '@mongodb-js/compass-components';
 
 import type { RootState } from '../../modules';
+import {
+  fetchNextPage,
+  fetchPrevPage,
+  cancelAggregation,
+} from '../../modules/aggregation';
+
+type PipelineResultsWorkspace = {
+  documents: Document[];
+  page: number;
+  perPage: number;
+  loading: boolean;
+  isPrevDisabled: boolean;
+  isNextDisabled: boolean;
+  error?: string;
+  onPrev: () => void;
+  onNext: () => void;
+  onCancel: () => void;
+};
+
+const topStyles = css({
+  display: 'flex',
+  justifyContent: 'flex-end',
+  alignItems: 'center',
+  gap: spacing[2],
+});
+
+const centeredContentStyles = css({
+  textAlign: 'center',
+  marginTop: spacing[4],
+});
+
+const errorMessageStyles = css({
+  color: uiColors.red.base,
+});
 
 const PipelineResultsWorkspace: React.FunctionComponent<PipelineResultsWorkspace> =
-  ({ documents }) => {
+  ({
+    documents,
+    page,
+    perPage,
+    loading,
+    isPrevDisabled,
+    isNextDisabled,
+    error,
+    onPrev,
+    onNext,
+    onCancel,
+  }) => {
+    const showingFrom = (page - 1) * perPage;
+    const showingTo = showingFrom + documents.length;
     return (
       <div data-testid="pipeline-results-workspace">
-        <pre>
-          <code>{JSON.stringify(documents, null, 2)}</code>
-        </pre>
+        <div className={topStyles}>
+          <Body>
+            Showing {showingFrom + 1} - {showingTo}
```

Comment:
```suggestion
            Showing {showingFrom + 1} – {showingTo}
```

This is a really minor thing, but en dashes look a bit nicer for ranges than hyphen-minus and are generally the right type of dash here :slightly_smiling_face: